bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/util/mapper/ActionInputsToConfigDescriptionParameters.java

```diff
@@ -0,0 +1,136 @@
+     */
+    public static @Nullable List<ConfigDescriptionParameter> map(List<Input> inputs) {
+        List<ConfigDescriptionParameter> configDescriptionParameters = new ArrayList<>();
+
+        for (Input input : inputs) {
+            ConfigDescriptionParameter parameter = ActionInputsToConfigDescriptionParameters.map(input);
+            if (parameter != null) {
+                configDescriptionParameters.add(parameter);
+            } else {
+                configDescriptionParameters = null;
+                break;
+            }
+        }
+
+        return configDescriptionParameters;
+    }
+
+    /**
+     * Maps an {@link Input} to a {@link ConfigDescriptionParameter}.
+     * <br>
+     * When adding new input types, remember to update {@link SerialisedInputsToActionInputs} as well!
+     *
+     * @param input the input to map to a config description parameter
+     * @return the config description parameter or null if the input parameter has an unsupported type
+     */
+    public static @Nullable ConfigDescriptionParameter map(Input input) {
+        boolean supported = true;
+        ConfigDescriptionParameter.Type parameterType = ConfigDescriptionParameter.Type.TEXT;
+        String defaultValue = null;
+        boolean required = false;
+        String context = null;
+        switch (input.getType()) {
+            case "boolean":
+                defaultValue = "false";
+                required = true;
+            case "java.lang.Boolean":
+                parameterType = ConfigDescriptionParameter.Type.BOOLEAN;
+                break;
+            case "byte":
+            case "short":
+            case "int":
+            case "long":
+                defaultValue = "0";
+                required = true;
+            case "java.lang.Byte":
+            case "java.lang.Short":
+            case "java.lang.Integer":
+            case "java.lang.Long":
+                parameterType = ConfigDescriptionParameter.Type.INTEGER;
+                break;
+            case "float":
+            case "double":
+                defaultValue = "0";
+                required = true;
+            case "java.lang.Float":
+            case "java.lang.Double":
+                parameterType = ConfigDescriptionParameter.Type.DECIMAL;
+                break;
+            case "java.lang.String":
+                break;
+            case "java.time.LocalDate":
+                context = "date";
+                break;
+            case "java.time.LocalTime":
+                context = "time";
+                break;
+            case "java.time.LocalDateTime":
+            case "java.time.ZonedDateTime":
+                context = "datetime";
+                break;
+            case "org.openhab.core.library.types.DecimalType":
+                parameterType = ConfigDescriptionParameter.Type.DECIMAL;
+                break;
+            case "org.openhab.core.library.types.QuantityType":
```

Comment:
I think we should define a new context `quantity` and set it here.

Reply:
You will then allow any unit in UI?
I believe you need either a unit or a dimension.

Reply:
> I believe you need either a unit or a dimension?

Dimension ... sorry I missed to mention that here.

Reply:
How do you want we provide the dimension? There is no field for that yet.

Reply:
Create a new field `dimension` for use with the `quantity` context.

When is `unit` and `unitLabel` used?

Reply:
One of them is certainly set when in config file you set attribute "unit" like there for example 

https://github.com/openhab/openhab-addons/blob/main/bundles/org.openhab.binding.netatmo/src/main/resources/OH-INF/config/config.xml#L30

Don't know which one and what is difference between both.

Reply:
I should run the appropriate REST API to see how fields are set.

And I do not remember if Main UI is displaying this unit. To be checked also.

Reply:
Unit label is also set by a a few channel types:

https://github.com/openhab/openhab-addons/blob/4651f5c101dc8fb3dada30805260852d3a602726/bundles/org.openhab.binding.gardena/src/main/resources/OH-INF/thing/bridge.xml#L25

Reply:
Ah ok

Reply:
Do we use them in Main UI?

Reply:
I can't find any occurence of `unitLabel` in the UI code, so no.
I see two options to implement the Quantity support in the UI here:

- A new context with a new field that holds the dimension. Allows the user to use any input in that dimension
- Use a pre-defined unit (the default unit for the dimension) and let the user only input the decimal.

The second option would be easier to implement, but might be problematic for units like dB. WDYT?

Reply:
> So we will set DECIMAL and provide a default unit, you will display the unit and let the user fill a decimal value. When calling the POST API, you will build a string containing the value and the unit.
 
Agreed, just one detail:
I would rather call the API with the decimal value then and handle the unit on the core side ... it feels wrong to define parameter type DECIMAL and then POST a TEXT.

Reply:
Ok